guides/source/upgrading_ruby_on_rails.md

```diff
@@ -1725,7 +1725,9 @@ Rails 4.0 no longer supports loading plugins from `vendor/plugins`. You must rep
 * Rails 4.0 has changed `serialized_attributes` and `attr_readonly` to class methods only. You shouldn't use instance methods since it's now deprecated. You should change them to use class methods, e.g. `self.serialized_attributes` to `self.class.serialized_attributes`.
 
 * When using the default coder, assigning `nil` to a serialized attribute will save it
-to the database as `NULL` instead of passing the `nil` value through YAML (`"--- \n...\n"`).
+to the database as `NULL`. Active Record 3.2 saved the `nil` value to the database differently:
+      * In Active Record 3.2, if the database column corresponding to the serialized attribute was nullable, the `nil` value through YAML was saved into the database as `"--- \n...\n"`. Active Record 3.2 returned `nil` when retrieving that value, and Active Record 4.0 will do the same.
+      * In Active Record 3.2, if the database column corresponding to the serialized attribute was not nullable, the `nil` value through YAML was saved into the database as `"--- \'\'\n"`. Active Record 3.2 returned `""` when retrieving that value. Active Record 4.0 will return `nil` instead.
```

Comment:
Wouldn't it be clearer to use the present tense for this? eg: "ActiveRecord 3.2 *saves* the " 
All these cases are referring to `3.2`, wouldn't it make more sense to describe:

- When it is nullable
- When it isn't nullable

Maybe something along these lines would work:

```
In Rails 4.0 assigning `nil` to a serialized attribute will save it to the database as `NULL`, and it will be retrieved as `nil`. This simplifies the behaviour of Rails 3.2, which saves the `nil` value to the database in two ways: when nullable, the `nil` value is saved into the database as `"--- \n...\n"` and returned as `nil`, and non-nullable values are saved into the database as `"--- \'\'\n"`, and retrieved as `""`, so you no longer have to check for empty strings when querying non-nullable values after upgrading.
```

Reply:
@peterb  Great feedback. I like you wording better than mine :) Does @rafaelfranca agree? I'm not sure of who needs to approve what for this exactly.